app/src/main/java/com/example/tasky/feature_agenda/presentation/Agenda.kt

```diff
@@ -260,10 +270,15 @@ fun AgendaContent(
                     verticalArrangement = Arrangement.spacedBy(8.dp)
                 ) {
                     if (!agendaItems.isNullOrEmpty()) {
-                        items(
-                            items = agendaItems,
-                            key = { it.id }
-                        ) { agendaItem ->
+                        itemsIndexed(
+                            items = agendaItems
+                        ) { index, agendaItem ->
+                            if (index > 0 && agendaItem.startTime.toLocalDate() == needlePosition.toLocalDate() && agendaItem.startTime.isAfter(
+                                    needlePosition
+                                ) && agendaItems[index - 1].startTime.isBefore(needlePosition)
```

Comment:
What if all agenda items are later than now? Then the needle would have to show up as the first element, but the first `index > 0` check already prevents this or am I wrong?